src/Control/Distributed/Process/Tests/Closure.hs

```diff
@@ -363,19 +378,26 @@ testSpawnSupervised TestTransport{..} rtable = do
     [node1, node2]       <- replicateM 2 $ newLocalNode testTransport rtable
     [superPid, childPid] <- replicateM 2 $ newEmptyMVar
     thirdProcessDone     <- newEmptyMVar
+    linkUp               <- newEmptyMVar
 
-    forkProcess node1 $ do
+    p <- forkProcess node1 $ do
       us <- getSelfPid
       liftIO $ putMVar superPid us
-      (child, _ref) <- spawnSupervised (localNodeId node2) (waitClosure 1000000)
-      liftIO $ do
-        putMVar childPid child
-        threadDelay 500000 -- Give the child a chance to link to us
-        throw supervisorDeath
+      (child, _ref) <- spawnSupervised (localNodeId node2)
+                                       (sendPidClosure us `seqCP` $(mkStaticClosure 'expectUnit))
+      _ <- expect :: Process ProcessId
+
+      liftIO $ putMVar childPid child
+      -- Give the child a chance to link to us
+      _ <- expect :: Process ()
+      throw supervisorDeath
 
     forkProcess node2 $ do
       [super, child] <- liftIO $ mapM readMVar [superPid, childPid]
       ref <- monitor child
+      -- because monitor message was sent before message to process
+      -- we hope that it will be processed before
+      send p ()
```

Comment:
Monitor sends a message to the local NC:
```Haskell
monitor' :: Identifier -> Process MonitorRef
monitor' ident = do
  monitorRef <- getMonitorRefFor ident
  sendCtrlMsg Nothing $ Monitor monitorRef
  return monitorRef
```
I don't think you are earning any ordering guarantees.

There is one other place in tests where waiting for monitoring is [done](https://github.com/haskell-distributed/distributed-process-tests/blob/5e426233313a532c237f2b22edce5d8d1fa1b916/src/Control/Distributed/Process/Tests/CH.hs#L120).